### PR TITLE
Upgrade to Dotnet 8TLS

### PR DESCRIPTION
**Pull request by msalemcode**
# Description

This PR updates the SaaS Accelerator project to use .NET 8 LTS. This upgrade ensures that the project benefits from the latest features, performance improvements, and security updates provided by .NET 8 LTS.

## Changes Made

1. **Project File Updates**:
   - Updated the target framework in the project files to `net8.0`.

2. **Package Updates**:
   - Updated NuGet packages to versions compatible with .NET 8 LTS.

3. **Code Adjustments**:
   - Made necessary code changes to ensure compatibility with .NET 8 LTS.
   - Refactored deprecated APIs and replaced them with their .NET 8 LTS equivalents.

4. **Configuration Updates**:
   - Updated configuration files, such as `appsettings.json` and `appsettings.test.json`, to align with .NET 8 LTS best practices.

## Impact

- **Performance Improvements**: Leveraging the performance enhancements in .NET 8 LTS.
- **Security Enhancements**: Utilizing the latest security features and updates in .NET 8 LTS.
- **Long-Term Support**: Ensuring the project is on a supported version of .NET for the foreseeable future.

## Testing

- **Unit Tests**: All existing unit tests have been run and updated where necessary to ensure compatibility with .NET 8 LTS.
- **Integration Tests**: Conducted integration tests to verify that all components of the SaaS Accelerator work seamlessly with .NET 8 LTS.
- **Manual Testing**: Performed manual testing of critical paths to ensure no regressions or issues were introduced.

## Conclusion

This PR upgrades the SaaS Accelerator to .NET 8 LTS, bringing in the latest improvements and ensuring long-term support. The update includes necessary code adjustments, package updates, and configuration changes to maintain compatibility and leverage the benefits of .NET 8 LTS.

### CODE REVIEW

File: deployment/Deploy.ps1
Comment (santhoshb-msft) [inline deployment/Deploy.ps1:1]: az webapp create -g $ResourceGroupForDeployment -p $WebAppNameService -n $WebAppNameAdmin  --runtime dotnet:6 --output $azCliOutput

@msalemcode  Can we update the appservice to use dotnet 8 please.

**Review by santhoshb-msft — COMMENTED** [review]
added comments

File: deployment/Deploy.ps1
Comment (msalemcode) [reply deployment/Deploy.ps1:1]: Done

**Review by fthorntonai — APPROVED** [review]
- [Approved] Unit Tests: All existing unit tests have been run and updated where necessary to ensure compatibility with .NET 8 LTS.
- [Approved ] Integration Tests: Conducted integration tests to verify that all components of the SaaS Accelerator work seamlessly with .NET 8 LTS.
- [Approved] Manual Testing: Performed manual testing of critical paths to ensure no regressions or issues were introduced.

**Review by v-ade — APPROVED** [review]
Everything looks good. Reviewed and okay.

**Review by v-ade — APPROVED** [review]

**Comment by fthorntonai** [comment]
- [Approved] Unit Tests: All existing unit tests have been run and updated where necessary to ensure compatibility with .NET 8 LTS.
- [Approved ] Integration Tests: Conducted integration tests to verify that all components of the SaaS Accelerator work seamlessly with .NET 8 LTS.
- [Approved] Manual Testing: Performed manual testing of critical paths to ensure no regressions or issues were introduced.
- [WIP] E2E Manual Testing: Performed E2E testing of 
           1. deploy PR to a new instance
           2. update an offer tech config to the details of the above instance
           3. run an e2e customer purchase exp , pending -> activate -> change plan -> meter -> scheduled meter -> unsubscribe

**Review by fthorntonai — APPROVED** [review]
3. Run an e2e customer purchase exp ,
- [ Verified ] pending
- [ Verified ] activate
- [ Verified ] customer change plan from portal - webhook testing
- update admin portal setting "AcceptSubscriptioUpdates" to true
- and try the change plan.
- [ Accepted ] meter
- [ Accepted ] scheduled meter
- [ Verified ] unsubscribe

**Review by santhoshb-msft — APPROVED** [review]
Tested By Adedayo and Fredrick and Myself. Approving this.

**Pull request merged**